pkg/services/system_intakes.go

```diff
@@ -109,3 +112,37 @@ func NewFetchSystemIntakeByID(
 	}
 
 }
+
+// NewSubmitSystemIntake is a service to submit system intake to CEDAR
+func NewSubmitSystemIntake(
+	submit func(intake *models.SystemIntake, logger *zap.Logger) (string, error),
+	save func(intake *models.SystemIntake) error,
+	logger *zap.Logger,
+) func(intake *models.SystemIntake, logger *zap.Logger) error {
+	return func(intake *models.SystemIntake, logger *zap.Logger) error {
+		if intake.AlfabetID.Valid {
+			err := &apperrors.CedarError{
```

Comment:
We need some way to represent an existing record in `apperrors`. My understanding is this should eventually translate into a 409.

Reply:
`AlfabetID` and `Status` have to come from the DB, not the request or else someone can overwrite them in an API request and trigger this code. It might even be cleaner to trigger a submit, then use the existing DB row as the data to submit, or else we're potentially dealing with DB conflicts when we go to write at the end. Pulling from the DB gives us a better assumption that the data is clean.

It might not be the most important right now, but we also need to start thinking about locking these operations per row somehow. It's a little weird with the API requests in the middle, but we should be able to use `SELECT FOR UPDATE` in postgres with the same outcome.

Reply:
I think the way @Igarfinkle and @christopherhuii and I have been talking about status was that the system intake status 'submitted' would be sent from the client. I wonder if we need some kind of state machine to make sure something can go from one state to another so that we're not accidentally back tracking or having other issues around status.